src/ts/mapboxMap.ts

```diff
@@ -220,20 +213,17 @@ module powerbi.extensibility.visual {
                     'line_string': true     // Lasso is overriding the 'line_string' mode
                 },
             });
-            console.log(this.settings.api.zoom)
-
 
-            // console.log('geocoder', this.geocoder)
+            this.geocoder = new MapboxGeocoder({
+                accessToken: this.settings.api.accessToken,
+                zoom: 10
+            })
 
             this.map.addControl(new mapboxgl.NavigationControl());
             this.map.addControl(this.draw, 'top-left');
             this.map.addControl(this.autoZoomControl);
 
-            this.geocoder = new MapboxGeocoder({
-                accessToken: this.settings.api.accessToken,
-                zoom: 10
-            })
-            document.querySelector('.geoDiv').appendChild(this.geocoder.onAdd(this.map));
+            document.querySelector('.map').appendChild(this.geocoder.onAdd(this.map));
```

Comment:
You can add this control with `map.addControl(geocoder, 'top-left');` or any other control position as opposed to adding it to the map div manually.

Reply:
@ryanbaumann I would like to place the geocoder at the top middle, and its not a supported location. I was following this example here https://www.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-outside-the-map/

Should I add it top left and then try to manipulate the .css to get it into the top middle?

Reply:
@samgehret ah, gotcha.  Yes then you're technique here is 👍 to place in a custom-positioned div.